server/neptune/workflows/internal/revision_setter/workflow.go

```diff
@@ -0,0 +1,176 @@
+package revisionsetter
+
+import (
+	"context"
+	"time"
+
+	"github.com/docker/docker/pkg/fileutils"
+	"github.com/pkg/errors"
+	key "github.com/runatlantis/atlantis/server/neptune/context"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/activities"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/activities/github"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/activities/terraform"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/config/logger"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/metrics"
+	"go.temporal.io/sdk/temporal"
+	"go.temporal.io/sdk/workflow"
+)
+
+const (
+	GithubRetryCount    = 3
+	StartToCloseTimeout = 10 * time.Second
+)
+
+type Request struct {
+	Repo github.Repo
+	Root terraform.Root
+}
+
+type revisionSetterActivities interface {
+	SetPRRevision(ctx context.Context, request activities.SetPRRevisionRequest) (activities.SetPRRevisionResponse, error)
+}
+
+type githubActivities interface {
+	ListPRs(ctx context.Context, request activities.ListPRsRequest) (activities.ListPRsResponse, error)
+	ListModifiedFiles(ctx context.Context, request activities.ListModifiedFilesRequest) (activities.ListModifiedFilesResponse, error)
+}
+
+func Workflow(ctx workflow.Context, request Request) error {
+	// GH API calls should not hit ratelimit issues since we cap the TaskQueueActivitiesPerSecond for the min revison setter TQ such that it's within our GH API budget
+	ctx = workflow.WithActivityOptions(ctx, workflow.ActivityOptions{
+		StartToCloseTimeout: StartToCloseTimeout,
+		RetryPolicy: &temporal.RetryPolicy{
+			MaximumAttempts: GithubRetryCount,
```

Comment:
This doesn't just impact github, this also impacts the buildnotify call since you're not creating a new ctx.

Reply:
Yup, I think we should be fine with configuring buildnotify calls to 3 tries as well since we're the only consumer for this new API and if we fail after 3 attempts, the API is likely unavailable. I've changed the const name and added comment. Lmk if you think we should configure buildnotify calls differently